### PR TITLE
spr 1.3.3 (new formula)

### PR DESCRIPTION
**Pull request by sven-of-cord**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/spr.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    tag:      "v1.3.3",
+    revision: "1f6d99ef268643cead1341cfc51e57b84c50b962"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "pkg-config" => :build
+    depends_on "zlib"
+  end
```
Comment (carlocab) [inline Formula/spr.rb:14]:
```suggestion
  depends_on "rust" => :build
  uses_from_macos "zlib"

  on_linux do
    depends_on "pkg-config" => :build
  end
```

File: Formula/spr.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match "spr #{version}", shell_output("#{bin}/spr --version")
```
Comment (carlocab) [inline Formula/spr.rb:21]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (sven-of-cord) [reply Formula/spr.rb:21]: The difficulty is that you need GitHub credentials to do anything with this command line tool. Calling it with `--version` or `--help` are the only non-interactive operations you can do that does not require you to feed it a GitHub access token.

File: Formula/spr.rb (hunk 1 above)
Comment (sven-of-cord) [reply Formula/spr.rb:14]: I will make this change :+1:

File: Formula/spr.rb (hunk 2 above)
Comment (carlocab) [reply Formula/spr.rb:21]: Well, we can also do something that requires a GitHub token without actually providing one and then check that we get the expected error (exit code + error message).

**Pull request closed without merging**